### PR TITLE
Fix/ea 469 fix meetup country import

### PR DESCRIPTION
**Pull request by andrasguseo**
[EA-469]
1. Fix
When importing from Meetup, the country comes over as a 2-letter code, like "ch" for Switzerland, "us" for United States, etc, and that is how it will be stored in the database. However, for storing venues, we use and store the full country name, like "Switzerland", "United States" etc. 
Since there is no match to the 2-letter country code, the country dropdown will be empty when editing a venue.

2. Fix
If there is no state saved with the venue, then the state will not show up when editing a venue.

[EA-469]: https://stellarwp.atlassian.net/browse/EA-469?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Tribe/Aggregator/Record/Abstract.php [hunk 1]
```diff
@@ -1606,14 +1606,32 @@ public function insert_posts( $items = [] ) {
 					if ( isset( $item->venue->image ) ) {
 						$venue_data['FeaturedImage'] = $item->venue->image;
 					}
+					
+					// If the data is coming from Meetup, then fix the country.
+					// Meetup sends the country as a lowercase two-digit country code.
+					if (
+						$origin == 'meetup'
```
[the remark this answers is not in the document]
Comment (andrasguseo) [reply src/Tribe/Aggregator/Record/Abstract.php]: Fixed.

File: src/Tribe/Aggregator/Record/Abstract.php [hunk 2]
```diff
@@ -1606,14 +1606,32 @@ public function insert_posts( $items = [] ) {
...
+						&& isset( $item->venue->country )
+					) {
+						$event['Venue']['Country'] = $venue_data['Country'] = $this->maybe_fix_country( $item->venue->country );
+					}
+
+					// If "State" is empty, it will not show up on the venue editing screen.
+					if ( ! isset( $event['Venue']['State'] && isset( $item->venue->stateprovince ) ) ) {
```
[the remark this answers is not in the document]
Comment (andrasguseo) [reply src/Tribe/Aggregator/Record/Abstract.php]: Fixed.

File: src/Tribe/Aggregator/Record/Abstract.php [hunk 3]
```diff
@@ -1606,14 +1606,32 @@ public function insert_posts( $items = [] ) {
...
+						$event['Venue']['State'] = $venue_data['State'] = $item->venue->stateprovince;
+					}
+
+					// This will help prevent a superfluous Geocoding request in
+					// events-calendar-pro/src/Tribe/Geo_Loc.php:712
```
[the remark this answers is not in the document]
Comment (andrasguseo) [reply src/Tribe/Aggregator/Record/Abstract.php]: Fixed.

File: src/Tribe/Aggregator/Record/Abstract.php [hunk 4]
```diff
@@ -1606,14 +1606,32 @@ public function insert_posts( $items = [] ) {
...
+					$event['Venue']['GeoAddress'] = $venue_data['GeoAddress'] = $venue_data['Venue'];
 
 					if ( $venue ) {
 						$venue_id                   = $event['EventVenueID'] = $venue_data['ID'] = $venue->ID;
 						$found_venues[ $venue->ID ] = $event['Venue']['Venue'];
 
 						// Here we might need to update the Venue depending on the main GlobalID
 						if ( 'retain' === $update_authority_setting ) {
-							// When we get here we say that we skipped an Venue
+							// When we get here we say that we skipped a Venue
```
[the remark this answers is not in the document]
Comment (andrasguseo) [reply src/Tribe/Aggregator/Record/Abstract.php]: Fixed.

File: src/Tribe/Aggregator/Record/Abstract.php [hunk 5]
```diff
@@ -2832,4 +2846,31 @@ public function is_polling() {
 	public function generate_next_batch_hash() {
 		return md5( uniqid( '', true ) );
 	}
+
+	/**
+	 * Changes the two-digit country code to the country name.
+	 * Meetup sends the country as a two-digit country code, which the Venues post type cannot recognize.
+	 * It is also better/safer for other instances where the full country name might differ in the source vs. TEC.
+         *
```
[the remark this answers is not in the document]
Comment (andrasguseo) [reply src/Tribe/Aggregator/Record/Abstract.php]: Fixed.

File: src/Tribe/Aggregator/Record/Abstract.php [hunk 6]
```diff
@@ -2832,4 +2846,31 @@ public function is_polling() {
...
+         * @TODO Would this be better in Common?
```
[the remark this answers is not in the document]
Comment (andrasguseo) [reply src/Tribe/Aggregator/Record/Abstract.php]: Fixed.

[the remark this answers is not in the document]
Comment (andrasguseo) [reply src/Tribe/Aggregator/Record/Abstract.php]: Fixed.

File: src/Tribe/Aggregator/Record/Abstract.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (andrasguseo) [reply src/Tribe/Aggregator/Record/Abstract.php]: Fixed.

File: src/Tribe/Aggregator/Record/Abstract.php [hunk 7]
```diff
@@ -1606,24 +1606,38 @@ public function insert_posts( $items = [] ) {
...
+					// Meetup sends the country as a lowercase two-digit country code.
+					if (
+						'meetup' == $origin
+						&& isset( $item->venue->country )
+					) {
+						$event['Venue']['Country'] = $venue_data['Country'] = $this->maybe_fix_country( $item->venue->country );
+					}
+
+					// If "State" is empty, it will not show up on the venue editing screen.
+					if ( ! isset( $event['Venue']['State'] && null !== $item->venue->stateprovince ) ) {
```
[the remark this answers is not in the document]
Comment (andrasguseo) [reply src/Tribe/Aggregator/Record/Abstract.php]: Fixed.

File: src/Tribe/Aggregator/Record/Abstract.php [hunk 8]
```diff
@@ -1616,23 +1726,42 @@ public function insert_posts( $items = [] ) {
...
+					}
+
+					// If "State" is empty, it will not show up on the venue editing screen.
+					if (
+						! isset( $event['Venue']['State'] )
+						&& isset( $item->venue->stateprovince )
+					) {
+						$event['Venue']['State'] = $venue_data['State'] = $item->venue->stateprovince; // phpcs:ignore Squiz.PHP.DisallowMultipleAssignments.Found
+					}
+
```
Comment (Camwyn) [inline src/Tribe/Aggregator/Record/Abstract.php:1747]: This is the meat of the change.

**Review by Camwyn — APPROVED** [review]

**Pull request merged**